pkg/functionconfig/types.go

```diff
@@ -188,6 +187,28 @@ func (s *Spec) GetRuntimeNameAndVersion() (string, string) {
 	}
 }
 
+func (s *Spec) GetHTTPPort() int {
+	if s.Triggers == nil {
+		return 0
```

Comment:
Is `0` the right answer? Sometimes it's used to get a free HTTP port. Maybe `-1`?

Reply:
Yes, 0 - means to get a free port. The idea here is either to get a free port or one defined by the user / a previous run